crdppf/static/js/Crdppf/main.js

```diff
@@ -295,13 +295,21 @@ Crdppf.init_main = function(lang) {
 
                         var selectedRadio = Ext.getCmp('extractRadioGroup').getValue();
 
-                        var url = [
-                            Crdppf.printReportCreateUrl,
-                            selectedRadio.inputValue,
-                            "/",
-                            Crdppf.currentProperty.attributes.id
-                        ].join('');
-
+                        if(Crdppf.pdfRenderEngine == 'crdppf_mfp'){
+                          var url = [
+                              Crdppf.printReportCreateUrl,
+                              selectedRadio.inputValue,
+                              "/",
+                              Crdppf.currentProperty.attributes.id
+                          ].join('');
+                        } else {
+                          var url = [
+                              Crdppf.printReportCreateUrl,
+                              selectedRadio.inputValue,
+                              "/pdf/",
+                              Crdppf.currentProperty.attributes.egrid
+                          ].join('');
+                        }
```

Comment:
This is ok and it works, but a good practise here would be to not repeat yourself (DRY). You could, for instance do something like that:
```javascript
var printParams = {}
if(Crdppf.pdfRenderEngine == 'crdppf_mfp') {
  printParams = {
    url: "/",
    id: "id"
  }
} else {
  printParams = {
    url: "/pdf/",
    id: "egrid"
  }
}

var url = [
  Crdppf.printReportCreateUrl,
  selectedRadio.inputValue,
  printParams.url,
  Crdppf.currentProperty.attributes[printParams.id]
].join('');
```
It's easier to understand what changes when `crdppf_mfp` mode is on of off...

Another good way to deal with this is to put that in the settings. In your .yaml:
```yaml
pdf_renderer: overwrite_me

print_params:
  crdppf_mfp:
    url: /
    id: id
...
```
And then access it directly in your .js:

```javascript
var url = [
  Crdppf.printReportCreateUrl,
  selectedRadio.inputValue,
  settings['pdf_render_engine']['url']
  settings['pdf_render_engine']['id']
].join('');
```

Reply:
thanks a lot. Very good suggestions! 👍